torch_xla/csrc/torch_util.cpp

```diff
@@ -11,9 +11,17 @@ namespace torch_xla {
 void SymIntElements::AddSymIntNodeElements(c10::SymInt& size) {
   if (size.is_symbolic()) {
     // c10::SymInt --(convert)--> c10::SymIntNode --(cast)-->
-    // lazy::SymIntNodeImpl
+    // PythonSymNodeImpl --> getPyObj() --> XLASymIntNodeImpl
     // --(get)--> lazy::NodePtr --(cast)--> lazy::DimensionNode
     c10::SymNode symbolicIntNode = size.toSymNodeImpl();
+    auto* pySymNode =
+        dynamic_cast<torch::impl::PythonSymNodeImpl*>(symbolicIntNode.get());
+    auto symNode = py::cast<c10::SymNode>(py::handle(pySymNode->getPyObj()));
```

Comment:
Thanks for the fix! I wonder if there is an simple way to avoid doing the multiple cast operations whenever we want a `XLASymNodeImpl` from a `c10::SymNode`.

Reply:
I don't think there is, but we can ask @ezyang to confirm. I believe the invariant Ed wanted is that any `SymNode` that comes from python must be a python object and is wrapped in `PythonSymNodeImpl`. Ed added bindings for `SymNode`, so `XLASymNodeImpl` can turn into a python object when we do `.shape[0]` in python, thanks to pybind.
`PythonSymNodeImpl` in this design is simply a container for that python object.